src/honey_badger/honey_badger.rs

```diff
@@ -106,6 +125,22 @@ where
         Ok(Step::default())
     }
 
+    /// Handles an epoch start announcement.
+    fn handle_epoch_started(&mut self, sender_id: &N, epoch: u64) {
+        self.update_remote_epoch(sender_id, epoch);
```

Comment:
Are `handle_epoch_started` and `update_remote_epoch` identical now? Then one of them can be removed.